src/spox/_adapt.py

```diff
@@ -73,6 +73,10 @@ def adapt_inline(
     source_version = max({v for d, v in node.opset_req if d in ("", "ai.onnx")})
     target_version = target_opsets[""]
 
+    # convert_version fails if the inlined model does not import the default domain
+    seen_domains = {prot.domain for prot in protos}
+    if not seen_domains & {"", "ai.onnx"}:
```

Comment:
We should probably make these sort of checks consistent across the codebase.

Reply:
On the other hand: In this case, we are only interested in the domains used in this particular subgraph. It seems rather fitting to check that here rather than elsewhere. It is also quite specific. We check the domain of the nodes in `protos` rather than those in `node.model` directly.

Reply:
I guess there two main things to consider here:

1. The one I had in mind was checking for the default domain, which is often checked by both `""` and `"ai.onnx"`.

2. On the other, there's finding all the opsets in a model (or node). As you noted, there's the subgraphs which I forgot about.

I think either could be more standardised, but whatever works, probably just works.

Reply:
I agree. There should be an obvious place where we get that kind of information. For now I think we may file this under "refactoring the build step" Todo-item.